onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -662,7 +662,10 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
 #endif
   } else if (provider_name == onnxruntime::kDmlExecutionProvider) {
 #ifdef USE_DML
-    Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_DML(session_options, 0));
+    std::unordered_map<std::string, std::string> dml_options;
+    dml_options["PerformancePreference"] = "HighPerformance";
+    dml_options["DeviceFilter"] = "Gpu";
+    session_options.AppendExecutionProvider("DirectML", dml_options);
```

Comment:
can we elaborate on the test cases? there are a lot of different code paths in here - how will our tests exercise them?